arangodb-foxx-services/src/main/services/observed-reads-by-write.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2022 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { aql, db } from '@arangodb'
+
+
+/**
+ * Returns a list of execution events which reads are visible from the write of the given execution event
+ *
+ * @param writeEvent za.co.absa.spline.persistence.model.Progress
+ * @returns za.co.absa.spline.persistence.model.Progress[]
+ */
+export function observedReadsByWrite(writeEvent) {
+    return writeEvent && db._query(aql`
+        FOR wExPlan IN 1 OUTBOUND ${writeEvent} progressOf
```

Comment:
Missing `WITH` statement

Reply:
nit: capitalization

Reply:
Thanks, both fixed.
Surprisingly, it worked without `WITH` 😮

Reply:
when testing on a local ArangoDB set the `query.require-with=true`.
E.g. in the `arangod.conf` under the "query" section:
```properties
[query]
require-with = true
```